rubin_sim/maf/metrics/brownDwarfMetric.py

```diff
@@ -0,0 +1,124 @@
+import numpy as np
+import healpy as hp
+from .baseMetric import BaseMetric
+import rubin_sim.maf.utils as mafUtils
+from scipy import interpolate
+
+
+class BDParallaxMetric(BaseMetric):
+    """Calculate the distance to which one could reach a parallax SNR for a given object
+    Modification of ParallaxMetric. Taken from:
+    https://github.com/jgizis/LSST-BD-Cadence/blob/main/bd_allLT_baseline_17.ipynb
+
+    Uses columns ra_pi_amp and dec_pi_amp, calculated by the ParallaxFactorStacker.
+
+    Parameters
+    ----------
+    metricName : str, opt
+        Default 'parallax'.
+    m5Col : str, opt
+        The default column name for m5 information in the input data. Default fiveSigmaDepth.
+    filterCol : str, opt
+        The column name for the filter information. Default filter.
+    seeingCol : str, opt
+        The column name for the seeing information. Since the astrometry errors are based on the physical
+        size of the PSF, this should be the FWHM of the physical psf. Default seeingFwhmGeom.
+    mags : dict
+        The absolute magnitude of the obeject in question. Keys of filter name, values in mags.
+    distances : np.array
+        Distances to try putting the object at (pc).
+    atm_err : float, opt
+        The expected centroiding error due to the atmosphere, in arcseconds. Default 0.01.
+    badval : float, opt
+        The value to return when the metric value cannot be calculated. Default 0.
+
+    Good values for mags:
+    mags={'i': 16.00, 'z': 14.52, 'y': 13.58}, metricName='L0'
+    mags={'i': 16.41, 'z': 14.93, 'y': 13.97}, metricName='L1'
+    mags={'i': 16.73, 'z': 15.30, 'y': 14.33}, metricName='L2'
+    mags={'i': 17.4, 'z': 15.88, 'y': 14.89}, metricName='L3'
+    mags={'i': 18.35, 'z': 16.68, 'y': 15.66}, metricName='L4'
+    mags={'i': 18.71, 'z': 16.94, 'y': 15.87}, metricName='L5'
+    mags={'i': 19.27, 'z': 17.35, 'y': 16.27}, metricName='L6'
+    mags={'i': 20.09, 'z': 18.18, 'y': 17.13}, metricName='L7'
+    mags={'i': 20.38, 'z': 18.10, 'y': 17.04}, metricName='L8'
+    mags={'i': 20.09, 'z': 17.69, 'y': 16.57}, metricName='L9'
+    mags={'i': 20.22, 'z': 17.98, 'y': 16.77}, metricName='T0'
+    mags={'i': 21.10, 'z': 18.84, 'y': 17.45}, metricName='T1'
+    mags={'i': 21.97, 'z': 18.26, 'y': 16.75}, metricName='T2'
+    mags={'i': 22.50, 'z': 18.08, 'y': 16.50}, metricName='T3'
+    mags={'i': 22.50, 'z': 18.02, 'y': 16.32}, metricName='T4'
+    mags={'i': 22.69, 'z': 19.20, 'y': 17.43}, metricName='T5'
+    mags={'i': 23.00, 'z': 19.82, 'y': 18.06}, metricName='T6'
+    mags={'z': 21.17, 'y': 19.34}, metricName='T7'
+    mags={'z': 21.52, 'y': 19.75}, metricName='T8'
+    mags={'z': 21.82, 'y': 20.37}, metricName='T9'
```

Comment:
.. I am wondering if it would be helpful to make this dictionary be a little callable function in this class, perhaps something like:
```
def dwarfMags(dwarfType):
    dwarfDict = {}
    dwarfDict['L0'] = {'i': 16.00, 'z': 14.52, 'y': 13.58}
    dwarfDict['L1'] = {'i': 16.41, 'z': 14.93, 'y': 13.97}
    dwarfDict['L2'] = {'i': 16.73, 'z': 15.30, 'y': 14.33}
    dwarfDict['L3'] = {'i': 17.4, 'z': 15.88, 'y': 14.89}
    dwarfDict['L4'] = {'i': 18.35, 'z': 16.68, 'y': 15.66}
    dwarfDict['L5'] = {'i': 18.71, 'z': 16.94, 'y': 15.87}
    dwarfDict['L6'] = {'i': 19.27, 'z': 17.35, 'y': 16.27}
    dwarfDict['L7'] = {'i': 20.09, 'z': 18.18, 'y': 17.13}
    dwarfDict['L8'] = {'i': 20.38, 'z': 18.10, 'y': 17.04}
    dwarfDict['L9'] = {'i': 20.09, 'z': 17.69, 'y': 16.57}
    dwarfDict['T0'] = {'i': 20.22, 'z': 17.98, 'y': 16.77}
    dwarfDict['T1'] = {'i': 21.10, 'z': 18.84, 'y': 17.45}
    dwarfDict['T2'] = {'i': 21.97, 'z': 18.26, 'y': 16.75}
    dwarfDict['T3'] = {'i': 22.50, 'z': 18.08, 'y': 16.50}
    dwarfDict['T4'] = {'i': 22.50, 'z': 18.02, 'y': 16.32}
    dwarfDict['T5'] = {'i': 22.69, 'z': 19.20, 'y': 17.43}
    dwarfDict['T6'] = {'i': 23.00, 'z': 19.82, 'y': 18.06}
    dwarfDict['T7'] = {'z': 21.17, 'y': 19.34}
    dwarfDict['T8'] = {'z': 21.52, 'y': 19.75}
    dwarfDict['T9'] = {'z': 21.82, 'y': 20.37}
    if dwarfType in dwarfDict:
        return dwarfDict[dwarfType]
    else:
        warnings.warn(f'Could not find dwarf type {dwarfType} in lookup table')
        return {}
```